### ISSUE TITLE
Omit extensions from cabal file which are problematic for "stack ghci"

### ISSUE THREAD

**Issue by mgsloan** [yesodweb/yesod-scaffold#109]
`-XNoImplicitPrelude` can cause issues for modules which require the Prelude.  So, if we use "stack ghci" to lload a yesod project along with some other project, it doesn't work.  See this issue: https://github.com/commercialhaskell/stack/issues/1363

`-XTemplateHaskell` can also cause issues due to it changing the meaning of `f $x` and `f $(x)`.

I realize that this would be a bit of an unpleasant refactoring, but ti'd certainly be an improvement, at least with our current tools.  Ideally, we wouldn't need to do this - https://ghc.haskell.org/trac/ghc/ticket/10827 - so this could be considered an upstream issue.

**Comment by gregwebs** [comment]
I was considering switching to [base-noprelude](https://hackage.haskell.org/package/base-noprelude). I am not sure if that will somehow have other ripple effects in this situation, but it seems like it should work.

With respect to `TemplateHaskell` I have not seen it create issues. But I always have a space after the `$` application function and it seems peculiar not to.

**Comment by mgsloan** [comment]
Yeah, removing `-XTemplateHaskell` isn't nearly as important as `-XNoImplicitPrelude`, which pretty much always breaks things.

**Comment by snoyberg** [comment]
I know I felt quite differently in the past, but I'm now on board with
moving to in-module language extensions because of these tooling issues.

**Comment by gregwebs** [comment]
I take a more nuanced view on this issue: we can divide language extensions in to 2 categories based on whether they can break existing code or not.

Those that don't break anything (usually any syntax extension such as RecordWildCards and in general most extensions that are not about the type system) should go in the `.cabal` file.
Any extensions which can break other modules should go in the module itself. Many type-system additions could potentially alter type inference and thus break otherwise working code.

There is definitely a grey area. I would argue that everyone should be putting a space after the dollar operator. However, having TemplateHaskell on can still make error messages worse in some cases even if it doesn't alter the ability to compile the code.

**Comment by snoyberg** [comment]
I like that division.

**Comment by gregwebs** [comment]
2 pull requests above to move things in this direction (#110 and #111)
Also a stackage PR: https://github.com/fpco/stackage/pull/990

**Comment by snoyberg** [comment]
Given improvements in `stack ghci` since this was open, is this still relevant? I believe we handle the case of extensions in .cabal files better than we used to (though, as someone who's not an ardent `stack ghci` user, I could be wrong).
